internal/controllers/framework_controller.go

```diff
@@ -59,7 +59,16 @@ func (r *FrameworkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	return ctrl.Result{}, err
 }
 
+func cleanAndUpdateMap(ns *map[string]string, current map[string]string) {
```

Comment:
how about 
```golang
func makeCopy(current map[string]string) map[string]string {
	result := make(map[string]string, len(current))
	for k, v := range current {
		result[k] = v
	}
	return result
}

namespace.Labels = makeCopy(framework.Spec.Labels)
namespace.Annotations = makeCopy(framework.Spec.Annotations)

```

Reply:
Thanks to the discussion we had on slack, I've implemented a different approach.